### DISCUSSION TITLE
Question about videos

### DISCUSSION THREAD

**Discussion by KingNordin** [lavalink-devs/Lavalink#1000] [Q&A]
I am not sure how to properly explain this, but I have this very bizarre bug.

`await node.Rest.GetTracksAsync(directLink, LavalinkSearchType.Plain)`

I use this code to get tracks from direct links.
Now the problem is that at times the links work and at times they don't.

For example this link:

https://rr3---sn-5hne6nsd.googlevideo.com/videoplayback?expire=1704133908&ei=tLCSZZfWGdTa6dsP_I-cmAk&ip=&id=o-AI4qLT1G8GSZZulJOo7NBYo0mJdy7UJ20U46poeQbmAK&itag=140&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=wN&mm=31%2C26&mn=sn-5hne6nsd%2Csn-4g5ednly&ms=au%2Conr&mv=m&mvi=3&pl=23&gcr=nl&initcwndbps=1870000&vprv=1&mime=audio%2Fmp4&gir=yes&clen=5165303&dur=318.959&lmt=1682061240865345&mt=1704111680&fvip=1&keepalive=yes&fexp=24007246&c=ANDROID_TESTSUITE&txp=4532434&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRAIgdcOvK9aF6PIIjJLbwHgVh7jFx6rHFGeH6Z4de1xvw3ICICy-auELYE-C-skG8OChiIPDyLLiqqtfjokJeBq6T0fq&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AAO5W4owRQIhAIpCLsEM8Cw7MGeOl-STyZKx30kt0vUQhjTyrTQOgyVdAiAlfGxyOKudXFPnTsuwgfItHuEYR_SshUK3tecBP98oOQ%3D%3D

This one doesn't work

This link does work:

https://rr5---sn-5hne6n6e.googlevideo.com/videoplayback?expire=1704134047&ei=P7GSZbSmJufV6dsP862w6AE&ip=&id=o-ALjgwmf7r9nCV-CTUuJf_LFPr98GqgQeSyJ57-S4QZQN&itag=140&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=b-&mm=31%2C26&mn=sn-5hne6n6e%2Csn-4g5lznl7&ms=au%2Conr&mv=m&mvi=5&pl=23&initcwndbps=1442500&vprv=1&mime=audio%2Fmp4&gir=yes&clen=8904338&dur=550.150&lmt=1684378739526965&mt=1704112155&fvip=3&keepalive=yes&fexp=24007246&c=ANDROID_TESTSUITE&txp=1311224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cxpc%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRQIgQg0jbf6DI8UJ1DV6JjG6lnmEIXoI5GJjt0BUj1G-N8ICIQDsNM_zjl6looaxavumH2CWH_zthB8KibZr0C8ikmhuQQ%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AAO5W4owRgIhAM4Sr4_y1tKUeh6KqRstnFSzR43bsn4kMuQscrCHeMUcAiEA05ylf09j5FrQMLbK5K7i7iRHKvXnjcFYCrG2TVWNTpQ%3D

As you can see they are pretty similar, so I really don't understand why one works and the other one doesn't, could someone please help me with this?

**Comment by duncte123** [comment]
Those links expire, it's best you don't store them. Also your IP address is in them

**Reply by KingNordin** [reply]
I don't store them, I get an error of type LavalinkLoadResultType.LoadFailed. 
And I will remove the IP  address thank you

**Reply by duncte123** [reply]
It just means that google rejected the playback/loading of the link. Nothing that you can do about it. There's also a high chance that the file you're trying to stream is encrypted and that's why it fails to load it (if you're not decrypting it)

**Reply by KingNordin** [reply]
I am not decrypting the videos, it's strange tho cause both videos come from YouTube, 

`Uri youTubeUrl = new(youtubeVideoUrl, UriKind.Absolute);
var youtubeExplode = new YoutubeClient();
var streamInfoSet = await youtubeExplode.Videos.Streams.GetManifestAsync(ExtractVideoIdFromUri(youTubeUrl));
var audioStreamInfo = streamInfoSet.GetAudioOnlyStreams().OrderByDescending(a => a.Bitrate).Where(a => a.Container.Name == "mp4").FirstOrDefault();`

This is my current code for getting the YouTube video
So how would I go about decrypting such a video?

**Reply by duncte123** [reply]
Why aren't you just using the youtube audio source manager that ships with lavalink to play youtube tracks? 
I am barely knowledgeable about how youtube works but I do know that playing back the tracks via the URL is always going to be a headache if you don't know what you're doing.
